engine/access/rpc/backend/events_index.go

```diff
@@ -26,14 +26,22 @@ func NewEventsIndex(events storage.Events) *EventsIndex {
 	}
 }
 
+// Initialize replaces nil value with actual reporter instance
+// Expected errors:
+// - If the reporter was already initialized, return error
 func (e *EventsIndex) Initialize(indexReporter state_synchronization.IndexReporter) error {
```

Comment:
to add some context here. the indexer on ANs is responsible for indexing events, tx results, and registers. When the index is initially bootstrapped, it needs to load an execution state checkpoint from disk and index all of the data. This can take >1hr on some systems. As a result, this `Initialize` pattern was added to allow the Access API to start up and serve queries before the index is ready. During that time, all calls to get data from this struct should return `indexer.ErrIndexNotInitialized`, and the caller should handle that appropriately for the method.

Reply:
Understood. A reasonable approach, I just that it's documented properly so I could know this by checking the source code. Thanks.